### PR TITLE
Fix independent instrumentation

### PR DESCRIPTION
**Pull request by lachrist**
fixes https://github.com/getappmap/appmap-agent-js/issues/181

Fixing this was not straightforward because we essentially had to support distributed application. Fortunately, the architecture of the agent was already gear toward this goal. Still, I had to overhaul the `backend` and `receptor` component to better support sessions. Sessions ensure that the instrumenter in the jest transformer and the rest of the agent are sending their messages to the same trace in the backend.

### CODE REVIEW

File: components/trace/appmap/index.test.mjs [hunk 1]
```diff
@@ -11,7 +11,10 @@ const { undefined } = globalThis;
 const configuration = extendConfiguration(
   createConfiguration("protocol://host/home/"),
   {
-    name: "name",
+    appmap_dir: "appmap_dir",
+    appmap_file: "appmap_file",
+    "map-name": "map-name",
```
Comment (dividedmind) [inline components/trace/appmap/index.test.mjs:16]: This is a rather unfortunate key name. Is it dictated by some external constraint?

File: components/receptor/default/track.mjs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            sendBackend(backend, session, {
+              type: "stop",
+              track: record,
+              termination: { type: "manual" },
+            })
+          ) {
+            const maybe_trace = compileBackendTrace(backend, session, record);
+            assert(
+              maybe_trace !== null,
+              InternalAppmapError,
```
Comment (dividedmind) [inline components/receptor/default/track.mjs:70]: This argument looks like a duplicate.

**Review by dividedmind — APPROVED** [review]
I'm not sure I entirely understood the changes, but I appreciate the couple extra added comments :)

File: components/receptor/default/track.mjs (hunk 2 above)
Comment (lachrist) [reply components/receptor/default/track.mjs]: thansk, rebased

**Pull request merged**